hadoop-ozone/client/src/test/java/org/apache/hadoop/ozone/client/checksum/TestECFileChecksumHelper.java

```diff
@@ -0,0 +1,270 @@
+import org.apache.hadoop.hdds.scm.XceiverClientFactory;
+import org.apache.hadoop.hdds.scm.XceiverClientGrpc;
+import org.apache.hadoop.hdds.scm.XceiverClientReply;
+import org.apache.hadoop.hdds.scm.pipeline.Pipeline;
+import org.apache.hadoop.hdds.scm.pipeline.PipelineID;
+import org.apache.hadoop.ozone.client.MockOmTransport;
+import org.apache.hadoop.ozone.client.MockXceiverClientFactory;
+import org.apache.hadoop.ozone.client.OzoneBucket;
+import org.apache.hadoop.ozone.client.OzoneClient;
+import org.apache.hadoop.ozone.client.OzoneVolume;
+import org.apache.hadoop.ozone.client.rpc.RpcClient;
+import org.apache.hadoop.ozone.om.helpers.OmKeyInfo;
+import org.apache.hadoop.ozone.om.helpers.OmKeyLocationInfo;
+import org.apache.hadoop.ozone.om.helpers.OmKeyLocationInfoGroup;
+import org.apache.hadoop.ozone.om.helpers.ServiceInfoEx;
+import org.apache.hadoop.ozone.om.protocol.OzoneManagerProtocol;
+import org.apache.hadoop.ozone.om.protocolPB.OmTransport;
+import org.apache.hadoop.util.DataChecksum;
+import org.apache.hadoop.util.Time;
+import org.apache.ratis.thirdparty.com.google.protobuf.ByteString;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import static org.apache.hadoop.hdds.protocol.datanode.proto.ContainerProtos.ChecksumType.CRC32;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for ECFileChecksumHelper class.
+ */
+public class TestECFileChecksumHelper {
+  private OzoneClient client;
+  private RpcClient rpcClient;
+
+  @BeforeEach
+  public void init() throws IOException {
+    ConfigurationSource config = new InMemoryConfiguration();
+    OzoneClientConfig clientConfig = config.getObject(OzoneClientConfig.class);
+    clientConfig.setChecksumType(ContainerProtos.ChecksumType.CRC32C);
+
+    ((InMemoryConfiguration) config).setFromObject(clientConfig);
+
+    rpcClient = new RpcClient(config, null) {
+
+      @Override
+      protected OmTransport createOmTransport(
+          String omServiceId) {
+        return new MockOmTransport();
+      }
+
+      @Nonnull
+      @Override
+      protected XceiverClientFactory createXceiverClientFactory(
+          ServiceInfoEx serviceInfo) {
+        return new MockXceiverClientFactory();
+      }
+    };
+
+    client = new OzoneClient(config, rpcClient);
+  }
+
+  @AfterEach
+  public void close() throws IOException {
+    client.close();
+  }
+
+
+  @Test
+  public void testEmptyBlock() throws IOException {
```

Comment:
There is a same test in `TestReplicatedFileChecksumHelper`. Can we put them together and make it parameterized, since there are similar.

Reply:
Thank you for the review. So we can extract similar logic(e.g., create `OmKeyInfo` instance) used in `testEmptyBlock()` and `testOneBlock()` to be used by `TestReplicatedFileChecksumHelper` and `TestECFileChecksumHelper`?

Reply:
I mean change the TestReplicatedFileChecksumHelper#testEmptyBlock to ParameterizedTest. For example:
```
-  @Test
-  public void testEmptyBlock() throws IOException {
+  @ParameterizedTest
+  @ValueSource(strings = {"EC", "RATIS"})
+  public void testEmptyBlock(String type) throws IOException {
```
Follow above then we don't need  `TestECFileChecksumHelper`.